mplc/contributivity.py

```diff
@@ -938,103 +940,88 @@ def without_replacment_SMC(self, the_scenario, sv_accuracy=0.01, alpha=0.95):
-            self.contributivity_scores
-        )
-        end = timer()
-        self.computation_time_sec = end - start
+    # %% compute Partner value by reinforcement learning
 
-    def federated_SBS_quadratic(self, the_scenario):
+    def PVRL(self, the_scenario, learning_rate):
         start = timer()
-        logger.info(
-            "# Launching computation of perf. scores of"
-            "quadratic performance increase compared to previous collective model")
-
-        relative_perf_matrix = self.compute_relative_perf_matrix(the_scenario)
-        comp_rounds_kept = relative_perf_matrix.shape[0]
-
-        # Calculate contributivity score with quadratic importance function
-        contributivity_scores = np.array(np.square(np.arange(comp_rounds_kept))) \
-            .dot(np.nan_to_num(relative_perf_matrix))
-
-        # Return contributivity scores
-        self.name = "Federated step by step quadratic scores"
-        self.contributivity_scores = contributivity_scores
-        self.normalized_scores = self.contributivity_scores / np.sum(
-            self.contributivity_scores
+        w = np.zeros(the_scenario.partners_count)
+        partner_values = np.exp(w) / (1.0 + np.exp(w))
+        previous_partner_values = np.zeros(the_scenario.partners_count)
+        epsilon = 0.002
+
+        mpl = multi_partner_learning.MultiPartnerLearning(
+            the_scenario.partners_list,
+            1,
+            the_scenario.minibatch_count,
+            the_scenario.dataset,
+            the_scenario.multi_partner_learning_approach,
+            the_scenario.aggregation_weighting,
+            is_early_stopping=False,
+            is_save_data=True,
+            save_folder=the_scenario.save_folder,
+            init_model_from="random_initialization",
+            use_saved_weights=False,
         )
-        end = timer()
-        self.computation_time_sec = end - start
-
-    def federated_SBS_constant(self, the_scenario):
-        start = timer()
-        logger.info(
-            "# Launching computation of perf. scores of constant"
-            "performance increase compared to previous collective model")
-
-        relative_perf_matrix = self.compute_relative_perf_matrix(the_scenario)
-
-        # Calculate contributivity score as average of the relative performance for each round for each partner
-        contributivity_scores = np.nanmean(relative_perf_matrix, axis=0)
-
-        # Return contributivity scores
-        self.name = "Federated step by step constant scores"
-        self.contributivity_scores = contributivity_scores
+        mpl.compute_test_score()
+        previous_loss = mpl.loss_collective_models[-1]
+        t = 0
+        while (t < the_scenario.epoch_count * the_scenario.partners_count or np.sum(
+                np.abs(partner_values - previous_partner_values)) / the_scenario.partners_count > epsilon):
+            t += 1
+            logger.info("t:", t)
```

Comment:
Actually this only logs the first "t".
We should do something like:             `logger.info(f"t: {t}")`

Reply:
In fact this t represents the number of epoch done. I renamed it PVRL_epoch. Due to the implementation of PVRL, most of the log will make no sense. (as a new mpl object is recreated at each epoch). It will be solved with the refacto of mpl, but I need changes which are done in the refacto to modify that.